### PR TITLE
DELETE should return 204 because of no content

### PR DESCRIPTION
**Pull request by d2lam**
- `200` expects some content there, so we should return `204` instead
- Update old tests to promise form

### CODE REVIEW

File: plugins/pipelines/remove.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ module.exports = () => ({
                     })
 
                     // user has good permissions, remove the pipeline
-                    .then(() => reply(pipeline.remove()));
+                    .then(() => reply(pipeline.remove()).code(204));
```
Comment (stjohnjohnson) [inline plugins/pipelines/remove.js]: Shouldn't this be something like:
`.then(pipeline.remove).then(() => reply().code(204));`

File: test/plugins/pipelines.test.js [hunk 2]
```diff
@@ -290,9 +290,9 @@ describe('pipeline plugin test', () => {
             pipelineFactoryMock.get.withArgs(id).resolves(pipeline);
         });
 
-        it('returns 200 when delete successfully', (done) => {
+        it('returns 204 when delete successfully', (done) => {
             server.inject(options, (reply) => {
-                assert.equal(reply.statusCode, 200);
+                assert.equal(reply.statusCode, 204);
                 done();
```
Comment (stjohnjohnson) [inline test/plugins/pipelines.test.js]: plz test that delete was called

File: test/plugins/pipelines.test.js [hunk 3]
```diff
@@ -290,9 +290,9 @@ describe('pipeline plugin test', () => {
             pipelineFactoryMock.get.withArgs(id).resolves(pipeline);
         });
 
-        it('returns 200 when delete successfully', (done) => {
+        it('returns 204 when delete successfully', (done) => {
             server.inject(options, (reply) => {
```
Comment (stjohnjohnson) [inline test/plugins/pipelines.test.js]: can you use the promise form instead of the callback?

File: test/plugins/secrets.test.js [hunk 4]
```diff
@@ -267,9 +267,9 @@ describe('secret plugin test', () => {
             });
         });
 
-        it('returns 200 if remove successfully', (done) => {
+        it('returns 204 if remove successfully', (done) => {
             server.inject(options, (reply) => {
```
Comment (stjohnjohnson) [inline test/plugins/secrets.test.js]: can you use the promise form instead of the callback?

File: plugins/pipelines/remove.js (hunk 1 above)
Comment (d2lam) [inline plugins/pipelines/remove.js]: Fixed

File: test/plugins/pipelines.test.js (hunk 3 above)
Comment (d2lam) [inline test/plugins/pipelines.test.js]: 👍  I fixed other tests too

File: test/plugins/pipelines.test.js (hunk 2 above)
Comment (d2lam) [inline test/plugins/pipelines.test.js]: Added

**Pull request merged**